Allow search table names in mixed cases

coresh opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | improvement
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

Updated pull request

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers/cases - upper-camel-lower-title-mixed.

Reverse compatibility - observed.
The function will give the correct result for any table names registers/cases - upper-camel-lower-title-mixed.

Thank you

coresh committed:
Check table names in upper-camel-lower-title cases

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function
searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers.
Optional argument "$lowerCase = true" - allows to maintain compatibility with previous versions

Thank you

coresh committed:
Merge branch 'doctrine:3.6.x' into 3.6.x

coresh committed:
Allow table names in upper-camel-lower-title cases

Updated pull request

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers/cases - upper-camel-lower-title-mixed.

Reverse compatibility - observed.
The function will give the correct result for any table names registers/cases - upper-camel-lower-title-mixed.

Thank you

coresh committed:
Allow search table names in upper-camel-lower-mix cases

Updated pull request

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers/cases - upper-camel-lower-title-mixed.

Reverse compatibility - observed.
The function will give the correct result for any table names registers/cases - upper-camel-lower-title-mixed.

Thank you

coresh committed:
Merge branch 'doctrine:3.6.x' into coresh-patch-1

derrabus commented:
Apparently, your patch breaks a lot of tests. And in addition to that, you haven't added a test that covers the problem you're trying to solve. I'm afraid, I cannot accept this patch.

coresh committed:
Merge branch 'doctrine:3.6.x' into coresh-patch-1

coresh committed:
Allow search table names in mixed cases

<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | improvement
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

Updated pull request

Mysql / MariaDB / Persona allows to use the table name in different cases.

The table names might look like this:

UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename

However, the original implementation of the tablesExist() function searches for existing tables ONLY in lower case:
by using the function:
'strtolower()'

$names = array_map('strtolower', $names);

return count($names) === count(array_intersect($names, $tableNames));

The current pull request allows to search for table names in different registers/cases - upper-camel-lower-title-mixed.

Reverse compatibility - observed.
The function will give the correct result for any table names registers/cases - upper-camel-lower-title-mixed.

Thank you

coresh committed:
Merge pull request #1 from coresh/coresh-patch-2

Allow search table names in mixed cases
